metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -377,7 +377,19 @@ private ReplicationControlManager(
     }
 
     public void replay(TopicRecord record) {
-        topicsByName.put(record.name(), record.topicId());
+        Uuid existingUuid = topicsByName.put(record.name(), record.topicId());
+        if (existingUuid != null) {
+            // We don't currently support sending a second TopicRecord for the same topic name...
+            // unless, of course, there is a RemoveTopicRecord in between.
+            if (existingUuid.equals(record.topicId())) {
+                throw new RuntimeException("Found duplicate TopicRecord for " + record.name() +
+                        " with topic ID " + record.topicId());
```

Comment:
nit: Noting that this first clause is explicitly rejecting the possibility of an idempotent replay of the same record a second time.  This makes sense, as it is unexpected if it should happen. 
 It might be good to call this out explicitly in a comment.

Reply:
The comment says:
```
            // We don't currently support sending a second TopicRecord for the same topic name...
            // unless, of course, there is a RemoveTopicRecord in between.```

which I think does cover the scenario you were thinking of (let me know if this makes sense)